mods/mkrcraft/src/main/kotlin/pw/mkr/craft/ServerInit.kt

```diff
@@ -0,0 +1,71 @@
+package pw.mkr.craft
+
+import net.fabricmc.api.DedicatedServerModInitializer
+import net.minecraft.entity.player.PlayerEntity
+import net.minecraft.server.network.ServerPlayerEntity
+import net.minecraft.text.LiteralText
+import net.minecraft.world.GameMode
+import pw.mkr.craft.blocks.binding.BindingBlock
+import pw.mkr.craft.events.chunkentry.ChunkEntryEvent
+import pw.mkr.craft.events.chunkentry.ChunkEntryListener
+import pw.mkr.craft.models.Binding
+import pw.mkr.craft.models.Chunk
+import pw.mkr.craft.blocks.portals.PortalBlock
+import pw.mkr.craft.utils.StoreManager
+
+@Suppress("Unused")
+object ServerInit : DedicatedServerModInitializer {
+    const val MOD_ID = "mkrcraft"
+
+    override fun onInitializeServer() {
+        StoreManager.init()
+        BindingBlock.register()
+        PortalBlock.register()
+
+        ChunkEntryEvent.listen(object : ChunkEntryListener {
+            override fun onChunkEntry(player: PlayerEntity, oldChunk: Chunk, newChunk: Chunk): Boolean {
+                val oldBinding = StoreManager.chunkBoundTo(oldChunk)
+                val newBinding = StoreManager.chunkBoundTo(newChunk)
+
+                fun send(msg: String) = player.sendMessage(LiteralText(msg), false)
+                fun inform(action: String, isPlayerClaimer: Boolean, binding: Binding) =
+                    send(
+                        "$action ${
+                            if (isPlayerClaimer) "your"
+                            else "${binding.player}'s"
+                        } claim"
+                    )
+
+                player.server ?: return true
```

Comment:
Move this check up. Return early on `player.server ?: true` before doing any other logic.

Reply:
Actually, we don't need that at all, I'll just remove it